src/platform/silabs/platformAbstraction/WiseMcuSpam.cpp

```diff
@@ -104,6 +108,13 @@ void SilabsPlatform::StartScheduler()
 extern "C" {
 void sl_button_on_change(uint8_t btn, uint8_t btnAction)
 {
+#if SL_ICD_ENABLED
+    // If device is sleeping making sure we get only one release per button press
+    if (btnAction == BUTTON_RELEASED && btn0_pressed == false)
```

Comment:
This seems to be completely wrong. You have an API that returns the button number (0 or 1 ) and the corresponding action. there shouldn't be any need for an external global variable...